clippy_lints/src/operators/absurd_extreme_comparisons.rs

```diff
@@ -119,6 +119,19 @@ fn detect_absurd_comparison<'tcx>(
 }
 
 fn detect_extreme_expr<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>) -> Option<ExtremeExpr<'tcx>> {
+    // Detect Duration::from_secs(0), which is an extreme value because
+    // Duration's internal representation ensures it can never be less than zero
+    if let ExprKind::Call(_, args) = &expr.kind
+        && args.len() == 1
+        && is_type_diagnostic_item(cx, cx.typeck_results().expr_ty(expr), sym::Duration)
+        && let Some(Constant::Int(0)) = ConstEvalCtxt::new(cx).eval(&args[0])
```

Comment:
This is not robust enough, as this will match any function whose argument is 0 and returns a `Duration`, including a function such as:

```rust
    fn one_second_plus(n: u64) -> Duration {
        Duration::from_secs(n+1)
    }
```

which does not return a zero duration.